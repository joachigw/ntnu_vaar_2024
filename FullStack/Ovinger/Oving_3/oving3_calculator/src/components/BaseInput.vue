<template>
    <label v-if="label"> {{ label }}</label>
    <input
        v-bind="$attrs"
        :value="modelValue"
        :placeholder="label"
        @input="$emit('update:modelValue', $event.target.value)"
    />
</template>

<script>
export default {
    props: {
        label: {
            type: String,
            default: "",
        },
        modelValue: {
            type: [String, Number],
            default: "",
        },
    },
};
</script>

<style scoped>
label {
    font-size: 1.25rem;
}

input {
    border-radius: 10px;
    color: black;
    font-size: 1rem;
    padding: 1rem;
}
</style>
