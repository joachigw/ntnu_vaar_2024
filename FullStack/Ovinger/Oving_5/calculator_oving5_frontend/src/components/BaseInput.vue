<template>
    <label v-if="props.label"> {{ props.label }}</label>
    <input
        v-bind="$attrs"
        :value="props.modelValue"
        :placeholder="props.label"
        @input="$emit('update:modelValue', $event.target.value)"
    />
</template>

<script setup>
const props = defineProps({
    label: {
        type: String,
        default: "",
    },
    modelValue: {
        type: [String, Number],
        default: "",
    },
});
</script>

<style scoped>
label {
    font-size: 1.25rem;
}

input {
    border-radius: 10px;
    color: black;
    font-size: 1rem;
    padding: 1rem;
}
</style>
