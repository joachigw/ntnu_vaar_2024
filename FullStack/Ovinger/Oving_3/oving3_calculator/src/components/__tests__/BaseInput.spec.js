import { describe, it, expect } from "vitest";
import { mount } from "@vue/test-utils";
import BaseInput from "../BaseInput.vue";

describe("Base input field", () => {
  it("renders html correctly", () => {
    const wrapper = mount(BaseInput);
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("renders label, field and placeholder correctly", () => {
    const wrapper = mount(BaseInput, {
      props: {
        label: "testLabel",
      },
    });
    expect(wrapper.find("label").exists()).toBe(true);
    expect(wrapper.find("label").text()).toBe("testLabel");
    expect(wrapper.find("input").attributes("placeholder")).toBe("testLabel");
  });

  it("renders without label when prop isn't specified", () => {
    const wrapper = mount(BaseInput);
    expect(wrapper.find("label").exists()).toBe(false);
    expect(wrapper.find("input").attributes("placeholder")).toBe("");
  });

  it("binds modelValue prop correctly", async () => {
    const wrapper = mount(BaseInput, {
      props: {
        modelValue: "testValue",
      },
    });
    expect(wrapper.find("input").element.value).toBe("testValue");
    await wrapper.setProps({ modelValue: "Updated" });
    expect(wrapper.find("input").element.value).toBe("Updated");
  });

  it("emits update:modelValue event", async () => {
    const wrapper = mount(BaseInput);
    await wrapper.find("input").setValue("testValue");
    expect(wrapper.emitted("update:modelValue")).toBeTruthy();
    expect(wrapper.emitted("update:modelValue")[0][0]).toBe("testValue");
  });
});
