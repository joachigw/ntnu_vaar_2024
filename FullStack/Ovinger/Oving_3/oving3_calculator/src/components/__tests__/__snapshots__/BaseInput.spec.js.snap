// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Base input field > renders html correctly 1`] = `
"<!--v-if-->
<input data-v-82c69faf="" placeholder="">"
`;
